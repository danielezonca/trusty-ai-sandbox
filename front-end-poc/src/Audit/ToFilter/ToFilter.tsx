import React, { useCallback } from "react";
import DatePicker from "../../Shared/components/DatePicker/DatePicker";

type ToFilterProps = {
  toDate: string;
  minDate: string;
  onToDateUpdate: (date: string) => void;
};

const ToFilter = (props: ToFilterProps) => {
  const { toDate, minDate, onToDateUpdate } = props;
  const updateToDate = useCallback((date: string) => onToDateUpdate(date), [onToDateUpdate]);

  return (
    <DatePicker value={toDate} minDate={minDate} onDateUpdate={updateToDate} id="audit-to-date" label="Ending date" />
  );
};

export default ToFilter;
