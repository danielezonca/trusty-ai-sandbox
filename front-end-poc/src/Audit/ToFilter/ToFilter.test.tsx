import React from "react";
import { mount, shallow } from "enzyme";
import ToFilter from "./ToFilter";

const defaultProps = {
  toDate: "2020-04-01",
  minDate: "2020-01-01",
  onToDateUpdate: jest.fn(),
};

describe("To filter", () => {
  test("renders", () => {
    const wrapper = shallow(<ToFilter {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("handles date update", () => {
    const wrapper = mount(<ToFilter {...defaultProps} />);
    const toDate = "2020-05-01";
    wrapper.props().onToDateUpdate(toDate);

    expect(defaultProps.onToDateUpdate).toHaveBeenCalledTimes(1);
    expect(defaultProps.onToDateUpdate).toHaveBeenCalledWith(toDate);
  });
});
