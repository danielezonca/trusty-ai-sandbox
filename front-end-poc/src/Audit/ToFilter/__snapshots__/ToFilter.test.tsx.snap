// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To filter renders 1`] = `
<DatePicker
  id="audit-to-date"
  label="Ending date"
  minDate="2020-01-01"
  onDateUpdate={[Function]}
  value="2020-04-01"
/>
`;
